OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -65,7 +67,11 @@ public IEnumerable<Order> Order(World world, CPos xy, MouseInput mi)
 				world.CancelInputMode();
 
 			var ret = InnerOrder(world, xy, mi).ToArray();
-			if (ret.Length > 0)
+
+			// If there was a successful placement order
+			if (ret.Any(o => o.OrderString == "PlaceBuilding"
+				|| o.OrderString == "LineBuild"
```

Comment:
Personal preference (and I'm pretty sure we do this elsewhere in the code): can you move the `||`s to the end of the preceding line?